pkg/controllers/edgeconnect/secret/secret.go

```diff
@@ -46,6 +46,11 @@ func PrepareConfigFile(ctx context.Context, instance *edgeconnectv1alpha1.EdgeCo
 		cfg.RootCertificatePaths = append(cfg.RootCertificatePaths, consts.EdgeConnectMountPath+"/"+consts.EdgeConnectCustomCertificateName)
 	}
 
+	if instance.Spec.ServiceAccountName != "" {
+		cfg.RootCertificatePaths = append(cfg.RootCertificatePaths, consts.EdgeConnectServiceAccountCAPath)
+		cfg.Secrets = append(cfg.Secrets, createDefaultServiceAccount(token))
```

Comment:
The secret shall only be added if automation is enabled.

Reply:
fixed in 3fa49162